### PR TITLE
ENH: simple proxy support added

### PR DESCRIPTION
**Pull request by xkord**

### CODE REVIEW

**Comment by expectocode** [comment]
Thank you so much for the good work! Looking over the code, I have one quibble - you return None from parse_proxy_str in some error cases eg no scheme - it might be more suitable to raise some kind of config syntax error.

File: setup.py [hunk 1]
```diff
@@ -27,10 +27,11 @@
     keywords='Telegram messaging database',
     packages=find_packages(),
     install_requires=[
-        'tqdm', 'telethon-aio', 'appdirs',
+        'tqdm', 'telethon-aio', 'appdirs', 'PySocks',
```
Comment (Lonami) [inline setup.py]: Not sure if this should be a hard dependency. We could do something like `pip install telegram-export[proxy]`, so that it doesn't need to be installed unless you need it.

File: setup.py [hunk 2]
```diff
@@ -27,10 +27,11 @@
...
         'async_generator'  # Python 3.5 async gen support
     ],
     scripts=['bin/telegram-export'],
+    test_suite='telegram_export.tests',
```
Comment (Lonami) [inline setup.py]: Nice, may be better suited as a separate pull request perhaps?

**Edit**: nevermind, you added tests for the proxy, that's fine.

File: telegram_export/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import unittest
+import socks
+from telegram_export.utils import *
```
Comment (Lonami) [inline telegram_export/tests/test_utils.py]: Please avoid `import *` where possible, it's bad practice.

File: telegram_export/utils.py [hunk 4]
```diff
@@ -211,3 +213,38 @@ def action_to_name(action):
...
+    if proxy_type_str.lower() == "socks5":
+        proxy_type = socks.SOCKS5
+    elif proxy_type_str.lower() == "socks4":
+        proxy_type = socks.SOCKS4
+    elif proxy_type_str.lower() == "https":
+        proxy_type = socks.HTTP
+    elif proxy_type_str.lower() == "http":
+        proxy_type = socks.HTTP
+    else:
+        return None
```
Comment (Lonami) [inline telegram_export/utils.py]: Like @expectocode suggested, `raise ValueError(...)` may be more appropriated.

Comment (xkord) [reply telegram_export/utils.py]: Thanks, for good comments

File: telegram_export/utils.py [hunk 5]
```diff
@@ -2,6 +2,8 @@
 import mimetypes
 
 from telethon.tl import types
+from urllib.parse import urlparse
+import socks
```
Comment (Lonami) [inline telegram_export/utils.py]: Since it's an optional dependency you need to:
```python
try:
    import socks
except ImportError:
    socks = None
```

and…

File: telegram_export/utils.py [hunk 6]
```diff
@@ -211,3 +213,40 @@ def action_to_name(action):
         types.ChannelAdminLogEventActionToggleSignatures: 'toggle.signatures',
         types.ChannelAdminLogEventActionUpdatePinned: 'update.pinned',
     }.get(type(action), None)
+
+
+def parse_proxy_str(proxy_str):
```
Comment (Lonami) [inline telegram_export/utils.py]: …in this method you need `if not socks: raise`, advicing the user to install the dependency if they want to use a proxy.

**Comment by Lonami** [comment]
I won't bother more, fixed that tiny tiny thing for you. Looks good to me.

**Comment by expectocode** [comment]
This PR is very close to finished, nice job :+1: 

 I'd like the proxy configuration to also be possible in the config file, and the commandline argument to override that setting.

**Comment by expectocode** [comment]
Also, the config file should show some example acceptable config strings. You could use your test cases.

**Comment by xkord** [comment]
Ok, i've added this option to Dumper

**Comment by Lonami** [comment]
Let's not bother you more ;) thanks for the PR.
